gx_suite_definitions/gene_info.ipynb

```diff
@@ -0,0 +1,433 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# alias\n",
+    "validator.expect_column_values_to_be_of_type(\"alias\", \"list\")\n",
+    "validator.expect_column_values_to_not_be_null(\"alias\")\n",
+    "validator.expect_column_values_to_have_list_members_of_type(column=\"alias\", member_type=\"str\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# is_igap\n",
+    "validator.expect_column_values_to_be_of_type(\"is_igap\", \"bool\")\n",
+    "validator.expect_column_values_to_not_be_null(\"is_igap\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# is_eqtl\n",
+    "validator.expect_column_values_to_be_of_type(\"is_eqtl\", \"bool\")\n",
+    "validator.expect_column_values_to_not_be_null(\"is_eqtl\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# is_any_rna_changed_in_ad_brain\n",
+    "validator.expect_column_values_to_be_of_type(\"is_any_rna_changed_in_ad_brain\", \"bool\")\n",
+    "validator.expect_column_values_to_not_be_null(\"is_any_rna_changed_in_ad_brain\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# rna_brain_change_studied\n",
+    "validator.expect_column_values_to_be_of_type(\"rna_brain_change_studied\", \"bool\")\n",
+    "validator.expect_column_values_to_not_be_null(\"rna_brain_change_studied\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# is_any_protein_changed_in_ad_brain\n",
+    "validator.expect_column_values_to_be_of_type(\"is_any_protein_changed_in_ad_brain\", \"bool\")\n",
+    "validator.expect_column_values_to_not_be_null(\"is_any_protein_changed_in_ad_brain\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# protein_brain_change_studied\n",
+    "validator.expect_column_values_to_be_of_type(\"protein_brain_change_studied\", \"bool\")\n",
+    "validator.expect_column_values_to_not_be_null(\"protein_brain_change_studied\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# target_nominations\n",
+    "validator.expect_column_values_to_be_of_type(\"target_nominations\", \"str\")\n",
+    "with open(\"../src/agoradatatools/great_expectations/gx/json_schemas/gene_info/target_nominations.json\", \"r\") as file:\n",
+    "    target_nominations_schema = json.load(file)\n",
+    "validator.expect_column_values_to_match_json_schema(\"target_nominations\", json_schema=target_nominations_schema, mostly=0.98)"
```

Comment:
Do you have an example of one that doesn't match?

Reply:
Here is an example where `validation_study_details` is `null`:
```
      "[{\"source\": \"Community\", \"team\": \"Chang Lab\", \"rank\": \"PS1-7\", \"hgnc_symbol\": \"ICA1\", \"target_choice_justification\": \"ICA1 was identified as an important key driver by predictive network analysis in microglial cells of dorsolateral prefrontal cortex in ROSMAP cohort, and microglial cells of temporal cortex in MAYO cohort, and neuron cells of dorsolateral prefrontal cortex in ROSMAP cohort, and temporal cortex in MAYO cohort, and dorsolateral prefrontal cortex in ROSMAP cohort.\", \"predicted_therapeutic_direction\": \"Agonism predicted to reduce disease progression\", \"data_used_to_support_target_selection\": \"Predictive network analysis derived from RNAseq and GWAS genotypes\", \"data_synapseid\": \"syn18358612\", \"study\": \"Mayo, ROSMAP\", \"input_data\": \"Genetics, RNA\", \"validation_study_details\": null, \"initial_nomination\": 2019}]",
```
Here is one where `data_synapseid` is `null`:
```
      "[{\"source\": \"TREAT-AD\", \"team\": \"IUSM-Purdue\", \"rank\": null, \"hgnc_symbol\": \"BTK\", \"target_choice_justification\": \"By applying the algorithm and process developed in https://doi.org/10.1038/s41598-020-79740-x, we identified frequently co-expressed gene modules\", \"predicted_therapeutic_direction\": \"Antagonism predicted to reduce disease progression. Interact with inflammosome through NLRP3 activation; Frequently coexpressed in microglia core module from AD transcriptomic data\", \"data_used_to_support_target_selection\": \"AD and normal brain transcriptomic datasets\", \"data_synapseid\": null, \"study\": \"GSE5281, GSE48350, GSE33000, GSE44772, ROSMAP, MSBB, Mayo, ABA\", \"input_data\": \"RNA\", \"validation_study_details\": \"not prioritized for experimental validation\", \"initial_nomination\": 2022}]",

The invalid characters are harder to see in the print out but I know there are some in there because when I remove those constraints in the JSON schema less records fail the expectation.
```

Reply:
I think we can allow both `validation_study_details` and `data_synapseid` to be null here -- there are a lot of entries where this is true. I'm not sure what to do about invalid characters though other than ignore them...

Reply:
We can also make `study` nullable for now to reduce noise.

Reply:
@JessterB OK so I made `study`, `data_synapseid`, and `validation_study_details` nullable. There are now 2 records that fail because `data_used_to_support_target_selection` is null. Can I make that nullable as well?

As for the invalid character ones, I realized that my regex string I was using was also causing failures when a string had carriage returns (\r) and newlines (\n). I updated the string to allow those specific characters and now 0 records fail that expectation. Is it ok if those strings have carriage returns and newlines?

Reply:
* `data_used_to_support_target_selection` can be nullable
* CRs and newlines are ok

Reply:
I made these changes in the latest commit.